test/rules/dont_assign_param_reference.dart

```diff
@@ -0,0 +1,23 @@
+// Copyright (c) 2016, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+// test w/ `dart test/util/solo_test.dart dont_assign_param_reference`
+
+void badFunction(int parameter) { // LINT
+  parameter = 4;
+}
+
+void ok(String parameter) {
+  print(parameter);
+}
+
+class A {
+    void badFunction(int parameter) { // LINT
+    parameter = 4;
+  }
+
+  void ok(String parameter) {
+    print(parameter);
+  }
+}
```

Comment:
It would be good to test that it doesn't produce a lint when there is a local variable hiding a parameter:

void ok2(String parameter) {
  if (parameter == null) {
    int parameter = 2;
    parameter = 3;
  }
}